python-backend/app/api/mines/tailings/resources/tailings.py

```diff
@@ -45,48 +45,48 @@ def get(self, mine_tailings_storage_facility_guid=None):
     def post(self, mine_tailings_storage_facility_guid=None):
         if not mine_tailings_storage_facility_guid:
             data = self.parser.parse_args()
-            mine_guid = data['mine_guid']
+            mine_guid = data.get('mine_guid')
             # see if this would be the first TSF
             mine_tsf_list = MineTailingsStorageFacility.find_by_mine_guid(mine_guid)
             is_mine_first_tsf = len(mine_tsf_list) == 0
             mine_tsf = MineTailingsStorageFacility(
                 mine_guid=mine_guid,
-                mine_tailings_storage_facility_name=data['tsf_name'],
+                mine_tailings_storage_facility_name=data.get('tsf_name'),
                 **self.get_create_update_dict())
             db.session.add(mine_tsf)
             if is_mine_first_tsf:
                 try:
                     req_documents_url = get_documents_svc_url('/required?category=MINE_TAILINGS')
                     get_tsf_docs_resp = requests.get(
                         req_documents_url,
-                        headers={'Authorization': request.headers['Authorization']})
+                        headers={'Authorization': request.headers.get('Authorization')})
 
                     if get_tsf_docs_resp.status_code != 200:
                         self.raise_error(
                             500,
                             'get_tsf_req_docs returned error' + str(get_tsf_docs_resp.status_code))
 
-                    tsf_required_documents = get_tsf_docs_resp.json()['required_documents']
+                    tsf_required_documents = get_tsf_docs_resp.json().get('required_documents')
                     new_expected_documents = []
                     for tsf_req_doc in tsf_required_documents:
                         new_expected_documents.append({
                             'req_document_guid':
-                            tsf_req_doc['req_document_guid'],
+                            tsf_req_doc.get('req_document_guid'),
```

Comment:
same thing with these, this was a really quick and dirty way to make sure the other endpoint didn't change, cause if it did, this would crash, not just use None.... really strong unit tests will be needed to make sure a change to the json will not cause unexpected behavior.